File: 프로그래머스/level1_이상한 문자 만들기.py
def solution(s):
    answer = ''
    s_split = s.split(' ')
    for i in range(len(s_split)):
        s_list = list(s_split[i])
        for j in range(len(s_list)):
            if j % 2 == 0:
                s_list[j] = s_list[j].upper()
            else:
                s_list[j] = s_list[j].lower()
        s_split[i] = "".join(s_list)
    answer = " ".join(s_split)

    return answer

print(solution("try hello world"))


# def solution(s):
#     answer = ''
#     for word in s.split(" "):
#         n = ''
#         for i in range(len(word)):
#             if i%2 == 0 :
#                 n += word[i].upper()
#             else :
#                 n += word[i].lower()
#         answer += (n + " ")
#     return answer[0:-1]

File: 프로그래머스 복습(java)/src/level1/level1_이상한_문자_만들기.java
package level1;

import java.util.Locale;

public class level1_이상한_문자_만들기 {
    static class Solution {
        public String solution(String s) {
            String answer = "";
            String[] sp = s.split("");
            int x = 0;
            for (int i = 0; i < sp.length; i++) {
                if (sp[i].equals(" ")) {
                    x = 0;
                    answer += " ";
                } else {
                    if (x % 2 == 0) {
                        answer += sp[i].toUpperCase();
                    } else {
                        answer += sp[i].toLowerCase();
                    }
                    x += 1;
                }
            }
            return answer;

            //More Simple Code
//            String answer = "";
//            int cnt = 0;
//            String[] array = s.split(" ");
//            for (String ss : array) {
//                cnt = ss.contains(" ") ? 0 : cnt + 1;
//                answer += cnt % 2 == 0 ? ss.toLowerCase() : ss.toUpperCase();
//            }
//            return answer;
        }
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.solution("try hello world"));
    }
}
